server/src/main/java/org/elasticsearch/cluster/coordination/InstanceHasMasterHealthIndicatorService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.coordination;
```

Comment:
Should we place server `HealthIndicatorService` implementation next to the functionality they are checking or centrally in `org.elasticsearch.health`?

Reply:
I would prefer it lives together with it's component, that could allow a more intimate relationship and will be ultimately needed for healthindicator from plugins anyway.

Reply:
++